haec-user-service/src/app.module.ts

```diff
@@ -1,11 +1,21 @@
-import { Module } from '@nestjs/common';
+import { Module, NestModule, MiddlewareConsumer} from '@nestjs/common';
 import { UserModule } from './modules/user/user.module';
+import { UserService } from './modules/user/services/user.service';
+import { UserController } from './modules/user/controllers/user.controller';
 import { PrismaService } from './providers/prisma.service';
 import { ConfigModule } from '@nestjs/config';
+import { AuthenticationMiddleware } from './_middleware';
+
 
 @Module({
-  imports: [UserModule, ConfigModule.forRoot({ isGlobal: true, cache: true })],
+  imports: [UserModule, ConfigModule.forRoot({ isGlobal: true, cache: true }), UserModule],
```

Comment:
Is there a reason we have `UserModule` imported `twice`?

Reply:
No, this should only be imported once.